decimal_test.go

```diff
@@ -95,6 +126,15 @@ func TestNewFromStringErrs(t *testing.T) {
 		".5.2",
 		"8..2",
 		"8.1.",
+		"1e",
+		"1-e",
+		"1e9e",
+		"1ee9",
+		"1ee",
+		"1e1.2",
+		"123.456e1.3",
+		"1e-1.2",
+		"123.456e-1.3",
```

Comment:
You should also add this test `"123.456e" + strconv.FormatInt(math.MinInt64, 10)` here